server/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/AllocationDecider.java

```diff
@@ -104,4 +106,74 @@ public Decision canForceAllocatePrimary(ShardRouting shardRouting, RoutingNode n
             return decision;
         }
     }
+
+    /**
+     * Returns a {@link Decision} whether the given shard can be forced to the
+     * given node in the event that the shard's source node is being vacated.
+     * This allows nodes using a vacate-type node shutdown (replace/vacate) to
+     * override certain deciders in the interest of moving the shard away from
+     * a node that *must* be removed.
+     *
+     * It defaults to returning "YES" and must be overridden by deciders that
+     * opt-out to having their other NO decisions *not* overridden while vacating.
+     *
+     * The caller is responsible for first checking:
+     * - that a replacement/vacate is ongoing
+     * - the shard routing's current node is the source of the replacement
+     * - the node destination is the target of the replacement
```

Comment:
I added a comment to a resolved comment, duplicating here for completeness:

I think this last line should not be the responsibility of the caller, rather should be the responsibility of the `NodeReplacementAllocationDecider`.

Reply:
Do you mean something like 119e4c8d377e29f46a0bc6a5e69f71b2ab3c27fc ?

Reply:
Yes, that looks fine.

Reply:
But we should also remove the line from the comment here .